data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/GreaterThanOperator.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class GreaterThanOperator implements Operator<Boolean> {
+    @Override
+    public Integer getSymbol() {
+        return DataPrepperExpressionParser.GT;
+    }
+
+    @Override
+    public Boolean eval(Object... args) {
+        checkArgument(args.length == 2, "Operands length needs to be 2.");
+        if ((args[0] instanceof Integer) && (args[1] instanceof Float)) {
```

Comment:
You should be able to use the `Comparable` interface to simplify this. I don't believe it will work for comparing Integers and Floats. But, you can convert the Integer to a Float if one is a Float and the other isn't.

Also, you can improve on `args[0]` and `args[1]`:

```
Object leftValue = args[0];
Object rightValue = args[1];

// overly simplified code
return leftValue > rightValue;
```

Reply:
> I don't believe it will work for comparing Integers and Floats. But, you can convert the Integer to a Float if one is a Float and the other isn't.

@dlvenable Not sure if I follow this part. Do you mean sth as
```
Float.valueOf(leftValue) > Float.valueOf(rightValue)
```
or
```
Float.valueOf(leftValue).compareTo(Float.valueOf(rightValue))
```
?

Reply:
I adopted the proposed approach from Dinu below. Let me know your thoughts.

Reply:
One takeaway from @dlvenable comment is if either number is a float then both numbers can be type typecast as a float. Java will cast the int as a float internally in all cases where a float is part of an operation with an int.